src/lib/stitching/vortex/stitching.ts

```diff
@@ -292,6 +274,63 @@ export const vortexStitchingEnvironment = (localSchema: GraphQLSchema) => ({
           amount(_ => parent.totalCents).resolve({}, args),
       },
     },
+    AnalyticsRankedStats: {
+      entity: {
+        fragment: gql`
+          ... on AnalyticsRankedStats {
+            rankedEntity{
+              __typename
+              ... on AnalyticsArtwork{
+                entityId
+              }
+              ... on AnalyticsShow {
+                entityId
+              }
+              ... on AnalyticsArtist {
+                entityId
+              }
+            }
+          }
+        `,
+        resolve: (parent, _args, context, info) => {
+          const toSnakeCase = s =>
+            s
+              .replace(/\.?([A-Z])/g, (_x, y) => "_" + y.toLowerCase())
+              .replace(/^_/, "")
+          const getMPNameFromTypeName = (typename: string): string => {
+            switch (typename) {
+              case "AnalyticsArtwork":
+                return "Artwork"
+              case "AnalyticsArtist":
+                return "Artist"
+              case "AnalyticsShow":
+                return "PartnerShow"
+              default:
+                return ""
+            }
+          }
+          const typename = parent.rankedEntity.__typename
+          const fieldName = toSnakeCase(getMPNameFromTypeName(typename))
+          const id = parent.rankedEntity.entityId
+          return info.mergeInfo
+            .delegateToSchema({
+              schema: localSchema,
+              operation: "query",
+              fieldName,
+              args: {
+                id,
+              },
+              context,
+              info,
+              transforms: vortexSchema.transforms,
+            })
+            .then(response => {
+              response.__typename = getMPNameFromTypeName(typename)
+              return response
+            })
+        },
+      },
+    },
```

Comment:
Seeing as the number of types is limited, I wonder if it isn’t much more readable to just hardcode the values.

```js
          const [mpTypeName, mpFieldName] = () => {
	            switch (typename) {
	              case "AnalyticsArtwork":
	                return ["Artwork", "artwork"]
	              case "AnalyticsArtist":
	                return ["Artist", "artist"]
	              case "AnalyticsShow":
	                return ["PartnerShow", "partner_show"]
	              default:
	                return ""
	            }
	          })()
```

Reply:
Yes, and now that `PartnerShow` should be `Show` I can just do `const fieldname = typename.replace('Analytics', '').toLowercase()`

Reply:
made the change in https://github.com/artsy/metaphysics/pull/1770/commits/0e0bc4f826af7a2b17c3993e096c4fe9bc36c514

Reply:
Perfect

Reply:
Thanks for catching that. The deprecated messages are even less visible when referencing those fields in the code 😞